CI.hs

```diff
@@ -326,14 +326,21 @@ buildDists
     --      git fetch --tags && git submodule update --init --recursive
     -- and it won't be deleted between runs.
     if noGhcCheckout then
-      cmd "cd ghc && git clean -xdf && git submodule foreach git clean -xdf && git submodule foreach git checkout . && git checkout ."
+      cmd "cd ghc && git remote remove upstream || true && git clean -xdf && git submodule foreach git clean -xdf && git submodule foreach git checkout . && git checkout ."
```

Comment:
I'm not sure about the relative precedence of `||` vs `&&` in bash, iiuc you're trying to remove `upstream` but don't care if it's not there to begin with, but I think we should just split this into a few lines

Reply:
@akrmn  i'll follow up with a new diff that does that. mind landing this one as is for now please? (it's the most critical of the set you've reviewed today)

Reply:
never mind @akrmn! force pushed a revision that splits up and rebases on master. let's /azp run this one more time and assuming it's good, get it landed. thanks!